src/vpp_plugins/vpp-chaperone/VppConfigManager.cpp

```diff
@@ -1644,89 +1651,90 @@ void VppConfigManager::doCpePolicerConfig (VppClient &vppClient,
   // otherwise VPP will crash if policed traffic is being passed while the
   // config is deleted.
   bool ifaceStopped = false;
-  // Delete old tables
-  std::vector<u32> tableIds;
-  vppClient.getClassifierTableIds (tableIds);
+  u32 swIfIndex = vppClient.ifaceToVppIndex (interface);
+  u32 tableId;
+  vppClient.getClassifierByIndex (swIfIndex, tableId);
+
   VppClient::ClassifierTableConfig tableConfig;
-  for (const u32 &tableId : tableIds)
+  if (!ifaceStopped)
     {
-      if (!ifaceStopped)
-        {
-          ifaceStopped = true;
-          VLOG (1) << "Stopping interface "
-                   << interface << " before removing old policer config";
-          vppClient.setInterfaceFlags (interface, false);
-        }
-      tableConfig.tableIndex = tableId;
-      tableConfig.isAdd = 0; // Delete old table
-      vppClient.addDelClassifierTable (tableConfig);
+      ifaceStopped = true;
```

Comment:
This check is always going to trigger once and no longer makes sense. You can probably remove it. However if `policerConfig == false` and there is no classifier already on this interface, we should probably return (is this right @deltice ?)

Reply:
It can be okay to let the function continue if we care about cleaning up cases where classifier tables or policers might be configured using vppctl in some unexpected way.

If only using vpp_chaperone, it shouldn't be possible, for example, to have policers present while the classifier table is gone. It might be possible if someone uses vppctl to do things, I don't recall the exact combination that might lead to this though or what the behavior would be (maybe that policers exist and get shown by `vppctl show policer` but won't affect anything, but that could be confusing).

We still need to check if there is no classifier on the interface in either case. (To either return, or not stop the interface until we find a table/policer that does need to be deleleted.)

Reply:
Addressed Review Comment.